packages/ag-charts-enterprise/src/features/range-buttons/rangeButtons.ts

```diff
@@ -0,0 +1,116 @@
+import { _ModuleSupport, _Scene, _Util } from 'ag-charts-community';
+
+import { RangeButton } from './rangeButton';
+import { Toolbar } from './toolbar';
+
+const { BOOLEAN, ActionOnSet, Validate } = _ModuleSupport;
+
+export class RangeButtons extends _ModuleSupport.BaseModuleInstance implements _ModuleSupport.ModuleInstance {
+    @ActionOnSet<RangeButtons>({
+        changeValue(enabled) {
+            if (this.toolbar) this.toolbar.visible = enabled;
+        },
+    })
+    @Validate(BOOLEAN)
+    public enabled: boolean = false;
+
+    private y = 0;
+    private height = 30;
+    private margin = 10;
+
+    private activeButton?: string;
+    private buttonNodes = new Map<string, string>();
+    private buttonOffsetX = 0;
+    private buttonSpacingX = 10;
+
+    private toolbar = new Toolbar();
+
+    constructor(private readonly ctx: _ModuleSupport.ModuleContext) {
+        super();
+
+        this.toolbar.visible = this.enabled;
+
+        this.destroyFns.push(
+            ctx.scene.attachNode(this.toolbar),
+            ctx.interactionManager.addListener('hover', this.onHover.bind(this)),
+            ctx.interactionManager.addListener('click', this.onClick.bind(this)),
+            ctx.toolbarManager.addListener('button-added', this.onButtonAdded.bind(this)),
+            ctx.toolbarManager.addListener('button-removed', this.onButtonRemoved.bind(this))
+        );
+    }
+
+    async performLayout({ shrinkRect }: { shrinkRect: _Scene.BBox }): Promise<{ shrinkRect: _Scene.BBox }> {
+        if (this.enabled) {
+            const toolbarTotalHeight = this.height + this.margin;
+            shrinkRect.shrink(toolbarTotalHeight, 'top');
+            this.y = shrinkRect.y - this.height - this.margin;
+        } else {
+            this.y = 0;
+        }
+
+        return { shrinkRect };
+    }
+
+    async performCartesianLayout(opts: { seriesRect: _Scene.BBox }): Promise<void> {
+        const { x, width } = opts.seriesRect;
+
+        if (this.enabled) {
+            const { y, height } = this;
+            this.layoutNodes(x, y, width, height);
+        }
+    }
+
+    private onHover({ offsetX, offsetY }: _ModuleSupport.InteractionEvent<'hover'>) {
+        this.activeButton = undefined;
+
+        if (!this.enabled) return;
+
+        for (const button of this.toolbar.children) {
```

Comment:
The responsibilities seem a bit mixed here - I'm thinking perhaps `ToolbarManager` should be merged with this class in Community, but the only place we actually add buttons is from the Enterprise Zoom feature?